master/buildbot/worker/protocols/manager/msgpack.py

```diff
@@ -344,10 +355,38 @@ def get_message_result(self, msg):
         res1 = yield d
         return res1
 
+    @defer.inlineCallbacks
     def onConnect(self, request):
         if self.debug:
             log.msg(f"Client connecting: {request.peer}")
 
+        try:
+            value = request.headers['authorization']
+            username, password = decode_http_authorization_header(value)
+        except Exception as e:
+            raise ConnectionDeny(401, "Unauthorized") from e
+
+        try:
+            dispatcher = self.get_dispatcher()
+            yield dispatcher.master.initLock.acquire()
+
+            if username in dispatcher.users:
+                pwd, _ = dispatcher.users[username]
+                if pwd == password:
+                    self.worker_name = username
+                    authentication = True
+                else:
+                    authentication = False
+            else:
+                authentication = False
+        except Exception as e:
+            raise Exception("Internal error") from e
```

Comment:
catching bare exception do not add anything

Reply:
This exception catches any possible errors which are not related to authentication. Therefore, if an error appears, the adversary will not be able to see any additional information about the error so the attack surface is limited.